dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -422,11 +433,13 @@ class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink, public virtual
   typedef OPENDDS_MULTIMAP_CMP(RepoId, RtpsReader_rch, GUID_tKeyLessThan) RtpsReaderMultiMap;
   RtpsReaderMultiMap readers_of_writer_; // keys are remote data writer GUIDs
 
+  WriterSeqReadersMap writerBestEffortReaders_;
```

Comment:
Yeah, sometimes it’s a challenge to reflect the exact functionalities in a not-too-long variable name. I’ll try. We need a structure to map a write ID to the last sequence number and the associated best-effort readers. Let’s start from the general and move to the specific. 

A write ID maps to a sub-structure containing a sequence number and a reader ID set. The sub-structure is named as SeqReaders. 

Then how about naming the map as WriterToSeqReadersMap? 

Since the variable here is specifically for best-effort readers, its naming options may include:
writer_to_seq_best_effort_readers_map_;
writer_to_seq_best_effort_readers_;
writer_to_best_effort_readers_;

How about using the second one?
WriterToSeqReadersMap writer_to_seq_best_effort_readers_;

Reply:
Keep in mind that the sequence number is the writer’s information, not any reader’s information. The datalink uses it to keep track of the last sequence number that a writer sent, not that a reader received. The actual sequence number each reader last received could differ from this one and from one another, but should be less than this one.